File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/Membership/MultiAccountAccess.cs
using System;
using JetBrains.Annotations;
using Psi.Data.Models.Domain.Membership.MultiAccountAccess;

namespace Psi.Data.Models.ClientConfigurationModels.Membership
{
    public class MultiAccountAccess : SettingsBaseHelper
    {
        public MultiAccountAccess(ISettingsBase settingsBase) : base(settingsBase)
        {
        }

        [SettingKey("MultiAccountAccess.Enabled")]
        public bool Enabled
        {
            get => GetBoolValue();
            [UsedImplicitly] set => SetValue(value);
        }

        [SettingKey("MultiAccountAccess.MinVersion")]
        public double MinVersion
        {
            get => GetDoubleValue();
            [UsedImplicitly] set => SetValue(value);
        }

        [SettingKey("MultiAccountAccess.MobileEnabled")]
        public bool MobileEnabled
        {
            get => GetBoolValue();
            [UsedImplicitly] set => SetValue(value);
        }

        [SettingKey("MultiAccountAccess.MinimumAndroidVersion")]
        public string MinimumAndroidVersion
        {
            get => GetValue();
            [UsedImplicitly] set => SetValue(value);
        }

        [SettingKey("MultiAccountAccess.MinimumIosVersion")]
        public string MinimumIosVersion
        {
            get => GetValue();
            [UsedImplicitly] set => SetValue(value);
        }

        [SettingKey("MultiAccountAccess.PermissionSource")]
        public MultiAccountAccessPermissionSource PermissionSource
        {
            get => Enum.TryParse(GetValue(), true, out MultiAccountAccessPermissionSource permissionSource)
                ? permissionSource
                : MultiAccountAccessPermissionSource.Database;
            [UsedImplicitly] set => SetValue(value);
        }

        [SettingKey("MultiAccountAccess.LastNameMatching.NumberOfCharactersToMatch")]
        public int LastNameMatchingNumberOfCharactersToMatch
        {
            get => GetIntValue();
            [UsedImplicitly] set => SetValue(value);
        }

        [SettingKey("MultiAccountAccess.Disclosure.ShouldShowLink")]
        public bool DisclosureShouldShowLink
        {
            get => GetBoolValue();
            [UsedImplicitly] set => SetValue(value);
        }

        [SettingKey("MultiAccountAccess.GlobalFeatureAccess")]
        public string GlobalFeatureAccessEncrypted
        {
            get => GetValue();
            [UsedImplicitly] set => SetValue(value);
        }
    }
}

File: workspace/legacy-analyzer/output/infrastructure/models/Membership/MultiAccountAccess.ts
// Generated imports
import { MultiAccountAccessPermissionSource } from '../MultiAccountAccessPermissionSource';

export interface MultiAccountAccess {
    /** @settingKey MultiAccountAccess.Enabled */
    enabled: boolean;
    /** @settingKey MultiAccountAccess.MinVersion */
    minVersion: number;
    /** @settingKey MultiAccountAccess.MobileEnabled */
    mobileEnabled: boolean;
    /** @settingKey MultiAccountAccess.MinimumAndroidVersion */
    minimumAndroidVersion: string;
    /** @settingKey MultiAccountAccess.MinimumIosVersion */
    minimumIosVersion: string;
    /** @settingKey MultiAccountAccess.PermissionSource */
    multiAccountAccessPermissionSource: MultiAccountAccessPermissionSource;
    /** @settingKey MultiAccountAccess.LastNameMatching.NumberOfCharactersToMatch */
    lastNameMatchingNumberOfCharactersToMatch: number;
    /** @settingKey MultiAccountAccess.Disclosure.ShouldShowLink */
    disclosureShouldShowLink: boolean;
    /** @settingKey MultiAccountAccess.GlobalFeatureAccess */
    globalFeatureAccessEncrypted: string;
}
